app/services/manifest_builder.rb

```diff
@@ -453,13 +453,21 @@ def manifest_image_path(resource)
     # Retrieve the URL path for an image served over the Riiif
     # @param [String] id identifier for the image resource
     # @return [String]
-    def manifest_image_thumbnail_path(id)
-      file_set = query_service.find_by(id: Valkyrie::ID.new(id))
+    def manifest_image_thumbnail_path(id_or_resource)
+      file_set = if id_or_resource.is_a?(Valkyrie::Resource)
```

Comment:
I'm not a big fan of multi-line assignments — would you mind pulling that out to a method since it's used twice?